### ISSUE TITLE
Multiple peptides with the same retention time?

### ISSUE THREAD

**Issue by Hassan-1991** [bzhanglab/AutoRT#20]
Dear Concern,

I'm trying to use AutoRT's transfer learning to predict retention times for peptides in a bacterial proteomics dataset. When preparing the input files for training, I notice that my mgf files often report multiple, very different retention times for the same peptide (in the same experiment). How should I account for this when making the training dataset? Or is this a problem with the data that I'm working with?

I'd be grateful for your guidance in this regard!

Thanks and best regards,
Hassan

**Comment by wenbostar** [comment]
Hi Hassan,

First make sure all your identified peptides are confident, for example, passed 1% FDR. If all your peptides identified from the same run (i.e., a single raw MS/MS file) for training are confidently identified, you may still have some peptideforms (peptide sequence + modification) for which different spectra from the same peptide form have very different RTs. But based on what I have observed, the fraction of peptides like that is typically less than 5%. For the same peptideform with multiple MS/MS spectra, you can follow what we did in the [AutoRT paper](https://www.nature.com/articles/s41467-020-15456-w):

> peptides for which the difference between the maximum observed RT and minimum observed RT was >3 min were removed from the training data.

> the average of RTs from all spectra identified from the same run with 1% FDR at both PSM and peptide levels for a peptide to represent observed RT for the peptide.

Bo

**Comment by Hassan-1991** [comment]
Thanks so much for the super quick response Bo! Let me try this and get back to you.

Hassan

**Comment by Hassan-1991** [comment]
Dear Bo,

I implemented your recommendations, and the results seem to make sense. One issue, however, is that a lot more peptides in my data had multiple retention times (over 50%), even after accounting for modified peptidoforms and only limiting myself to those that passed the 0.01 q-value cutoff.

For full context, I wanted to briefly lay out the steps I took to get here to make sure there are no obvious red flags in my analysis (feel free to move this elsewhere if you think a lot of the detail here is irrelevant):

1 - Search raw files with [MSGF+percolator](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3975676/). Database included all annotated and non-annotated ORFs in E. coli (generated with [iPtgxDB](https://iptgxdb.expasy.org/))
2 - Test set = Top 1000 ranked peptides from annotated proteins, training set = all other peptides that passed the 0.01 q-value cutoff. Conducted transfer learning per the code provided in the "example" directory.
3 - To extract retention times, I annotated each spectrum in my mgf file with PSM information from the mzid file using [this tool](https://spectra-cluster-py.readthedocs.io/en/latest/tools/mgf_search_result_annotator.html), then added in their modification information by further parsing the mzid file.
4 - For peptides with multiple retention times, I got rid of those with a difference of 180 seconds or more between the highest and lowest RT values. This took out about 11% of the total spectra that passed the 0.01 q-value cutoff. For the rest, I averaged the retention times in case of multiple values.

More than 98% of the peptides in my training set (i.e. those that passed the q-value 0.01 cutoff) came from annotated proteins, which is expected.

My ultimate goal is to validate the non-annotated peptides in my dataset. To do this, I followed the recommendations of [this paper](https://elifesciences.org/articles/78772) - who also used AutoRT - to check how many of the 162 non-annotated peptides in the test set had an absolute difference between actual and predicted retention times more than the 95th percentile value in the annotated peptide set. The 95th percentile value for the annotated protein actual-predicted RT difference was 1102 seconds, and 36 out of the 162 (22.2%) of the non-annotated peptides passed this cutoff.

I'd be grateful for your comments!

Thanks and best regards,
Hassan

**Comment by wenbostar** [comment]
Hi Hassan,

Did you train a single model for each raw MS/MS file?

> One issue, however, is that a lot more peptides in my data had multiple retention times (over 50%), even after accounting for modified peptidoforms and only limiting myself to those that passed the 0.01 q-value cutoff.

This is very common since many peptides identified from the same raw file are likely to have multiple MS/MS spectra identified and each spectrum has different observed RTs.

> The 95th percentile value for the annotated protein actual-predicted RT difference was 1102 seconds, and 36 out of the 162 (22.2%) of the non-annotated peptides passed this cutoff.

You could validate your novel peptides using [PepQuery](https://www.nature.com/articles/s41467-023-37462-4)? It could help to remove potential false positives and you are likely to have higher percentage of novel peptides passed the RT cutoff after that validation.

**Comment by wenbostar** [comment]
For your trained models, do you see performance on testing data like below?

![image](https://github.com/bzhanglab/AutoRT/assets/5302110/2221de5f-dd57-41c1-8d67-0f77ccd3f385)

**Comment by Hassan-1991** [comment]
Dear Bo, thanks so much for your incredibly helpful feedback! I'll definitely look into PepQuery, and see how this can fit into my overall validation pipeline.

I made a quick Excel graph with my predicted and observed retention times in minutes (N=10964, R2=0.989). The only weird thing I'm noticing is the way the predicted retention times are reaching a "ceiling" around 256 minutes - I don't know if I should be concerned about that. Also, these values are just for peptides from annotated proteins - when I include the 162 non-annotated peptides that also passed the FDR cutoff, there's more scatter and the R2 value slips to 0.979 - which is what I'd expect since a lot of these noncanonical peptides are probably not real. If I confine my analysis just to the 162 noncanonical peptides, there's no correlation (R2=0.46) - again, as expected.
<img width="762" alt="image" src="https://github.com/bzhanglab/AutoRT/assets/67057211/7379c30a-aed2-475e-a00d-f7f42353d13b">

Please let me know if this makes sense! It's incredibly helpful to get such helpful feedback from you, especially seeing how daunting the field of proteome bioinformatics is.

Thanks again and very best regards,
Hassan

**Comment by wenbostar** [comment]
It looks like something is not processed correctly. Could you share a raw MS/MS file to me?

**Comment by Hassan-1991** [comment]
Yep! This plot was made with the data from just one experiment, which I downloaded from here - https://ftp.pride.ebi.ac.uk/pride/data/archive/2015/08/PXD002140/MURI_100_1a_03312014.raw

**Comment by wenbostar** [comment]
I made an example based on this raw file and the notebook is available at https://github.com/bzhanglab/AutoRT/blob/master/example/Experiment_specific_RT_prediction_using_AutoRT_MaxQuant_Ecoli.ipynb.

**Comment by wenbostar** [comment]
Please reopen this issue if you still have questions related to this.
